includes/model/Shifts_model.php

```diff
@@ -556,13 +556,22 @@ function Shift_update($shift)
     );
 }
 
+/**
+ * Get the next free shifts transaction id
+ */
+function Shift_get_next_transaction_id(): int
+{
+    return Db::selectOne('SELECT MAX(transaction_id) + 1 AS transaction_id FROM Shifts')['transaction_id'] ?? 1;
```

Comment:
As discussed I would suggest one of these options:
- Lock the shifts table during shift creation
- Use a uuid as shift transaction id
- Create a shifttransaction table and reference to it as shift transaction id

Reply:
UUIDs it is